Design_Doc_Examples/RAG_Q&A_for collaborative_work_platform.md

```diff
@@ -144,17 +144,92 @@ If users keep using it after some time
 
 ### **X. Integration**
 
-Embeddings db
-Chat UI
-Platform integration
-Parallel processing
-SLAs
+### **i. Embeddings Database**
 
-- **Key Takeaways:**
-    1. Integration is a continuous and essential process that ensures the success and longevity of ML systems. It requires careful planning, from API design to deployment and operation.
-    2. API design should prioritize simplicity and predictability, with a focus on creating interfaces that hide complexity while allowing for necessary customization and ensuring deterministic behavior.
-    3. The release cycle of ML systems presents unique challenges, necessitating a balance between agility and stability. Techniques like blue-green and canary deployments can facilitate safer updates and minimize disruptions.
-    4. Operational robustness is achieved not only through technical means such as CI, logging, and monitoring but also by addressing non-technical aspects like compliance and user data management. Overrides and fallbacks are critical for maintaining service continuity and adapting to changes or failures in real-time.
+This is one of the core components in the system for efficient document search and retrival.
+
+**i.i. Embeddings Generation**
+
+- **Query Embeddings:** Converts client's query into embeddings for further selecting nearest neighbours within the database.
+- **Document Embeddings:** Creates embeddings using a pre-trained BERT-based model for new documents. The model processes documents via API request, resulting in document ID, document metadata and embeddings. The original file is stored in Documents Storage with the same document ID to avoid overloading the vector DB.
+- **Updates:** Automatically updates embeddings when document version changes to maintain vectors relevance.
+
+**i.ii. Database Features**
+
+- A cloud and scalable database.
+- Supports nearest neighbor search, using cosine or Euclidean similarity.
+- The following fields are stored for further mapping with Documents Storage:
+  1. Document ID
+  2. Version number
+  3. Metadata storage (document title, author, creation date)
+  4. Embeddings representation 
+- Embeddings, metadata, and queries are encrypted to ensure security. Strict access control managment. 
+
+### **ii. Documents Storage**
+
+A scalable cloud service (e.g. AWS S3) for scalable storage and files managment for the following data types:
+1. Original files uploaded by clients, including their versions. The service returns a URL (Document ID) for each uploaded file, which is stored in the embeddings database metadata.
+2. Chat communications and response ratings, structured as JSON objects with fields for user queries, responses, timestamps and ratings.
+
+### **iii. Chat UI**
+
+An intuitive and responsive interface for clients to query and receive results.
+
+**Features.**
+  1. Clients can upload new documents, which automatically triggers embedding generation and storage.
```

Comment:
@luckydevvvil should we allow uploading new documents from Chat UI or better require them upload documents in a regular already existing processes?

Reply:
We assume that the current document upload process is integrated within the Chat UI